lessons/configuration.md

```diff
@@ -28,15 +28,36 @@ This lessons consists of the following exercises:
 
 ## 1. Why do we use configuration? (Stace)
 
-### Steps
+When we run our applications locally, we run them against a known environment. We know the location of all resources we need, and if the code is staying on our machines then we also know all of our secrets.
 
-1.
-2.
-3.
+During the 90's, when deploying small applications you could even take a good guess that the users running your machine would have a similar environment to your own.
 
-> 📝 **Tip** - < TIP >
+If they were running DOS you could assume that they had a C drive where you could store data. Copy your file to the machines that need to run the application and it just works.
 
-> 🔎 **Observation** - < OBSERVATION >
+That didn't work for larger applications though, and in this day and age it doesn't really work at all.
+
+As applications became larger, as they became more complex and distributed, and as they were deployed to more diverse environments, you - the developer - lost the knowledge of where and how the application was going to run.
+
+You could make a change to your application and recompile it for each environment that you deploy into - but that is both a lot of extra (manual) work, and introduces risks as what is deployed to test and production environments are all different applications. This is not a realistic approach.
+
+The solution to this is to introduce configuration settings to your application. Something that can be changed externally to your code to allow the same runtime to work in multiple different places. That is, use environment settings to separate the runtime values in your code from the compile time.
+
+Some examples of the type of data we want to separate are:
+
+* Connection strings
+* The execution mode of the application (dev, test, production etc)
+* API URLs
+* Service account details
+
+Another reason for configuration settings is security. Our code no longer lives on our machines. It lives in software repositories: Azure DevOps, GitHub etc.
+
+Putting sensitive information into these environments, even for private repositories is a security risk as it allows anyone with access to the repository to know sensitive information about all of your environments. If your repository is public the risk is even greater!
```

Comment:
```suggestion
Putting sensitive information into these systems, even for private repositories is a security risk as it allows anyone with access to the repository to know sensitive information about all of your environments. If your repository is public the risk is even greater!
```

Reply:
Changed to 'source control systems' to match the previous paragraph and be explicit abou the ststems we mean